support.cpp

```diff
@@ -37,47 +37,48 @@ stripper_game_t stripper_game;
 static const char*
 parse_map(const char* map, const char* entities)
 {
-    FILE *fp;
-    char path[256];
-
-    g_Stripper.SetEntityList(entities);
-
-    stripper_game.path_format(path,
-            sizeof(path),
-            "%s/%s/global_filters.cfg",
-            stripper_game.game_path,
-            stripper_game.stripper_cfg_path);
-    fp = fopen(path, "rt");
-    if (fp == NULL)
-    {
-        stripper_game.log_message("Could not find global filter file: %s", path);
-    }
-    else
-    {
-        fclose(fp);
-        g_Stripper.ApplyFileFilter(path);
-    }
-
-    stripper_game.path_format(path,
-            sizeof(path),
-            "%s/%s/maps/%s.cfg",
-            stripper_game.game_path,
-            stripper_game.stripper_cfg_path,
-            map);
-    fp = fopen(path, "rt");
-    if (fp)
-    {
-        fclose(fp);
-        g_Stripper.ApplyFileFilter(path);
-    }
-
-    return g_Stripper.ToString();
+	FILE* fp;
+	char path[256];
+
+	g_Stripper.SetEntityList(entities);
+
+	stripper_game.path_format(path,
+		sizeof(path),
+		"%s/%s/global_filters.cfg",
+		stripper_game.game_path,
+		stripper_game.stripper_cfg_path);
+	fp = fopen(path, "rt");
+	if (fp == NULL)
+	{
+		stripper_game.log_message("Could not find global filter file: %s", path);
+	}
+	else
+	{
+		fclose(fp);
+		g_Stripper.ApplyFileFilter(path);
+	}
+
```

Comment:
One iteration can be skipped by writing it like this:
`	const char_t* pPart = map;
	while(*(pPart = strchrnul(pPart, '_')) != '\0')
	{
		printf("result: %s\n", pPart);
		*pPart++;
	}`
In the for loop at the last iteration pPart points at the string end, fopen does unneeded work.

Reply:
Your loop produces one part less and wrong ones. From my REPL sample (map `ctf_2fort`) it just generates a `_2fort` token. My loop was very carefully analyzed after trial and error, and proper understanding. The only way it could be reasonably optimized is by unifying the base path construction outside the loop (so the preceding until the `maps` subfiles).

You fell into the easy mistake I also realized when I started coding it. As `strchrnul` returns a pointer to the next search character (`_`), that isn't the value we want for the next file name part, but everything between `map` and `pPart` instead.